pilot/pkg/autoregistration/controller.go

```diff
@@ -151,10 +151,12 @@ type HealthStatus = v1alpha1.IstioCondition
 // NewController create a controller which manages workload lifecycle and health status.
 func NewController(store model.ConfigStoreController, instanceID string, maxConnAge time.Duration) *Controller {
 	if features.WorkloadEntryAutoRegistration || features.WorkloadEntryHealthChecks {
-		maxConnAge := maxConnAge + maxConnAge/2
-		// if overflow, set it to max int64
-		if maxConnAge < 0 {
-			maxConnAge = time.Duration(math.MaxInt64)
+		if maxConnAge != math.MaxInt64 {
```

Comment:
very nit: I don't think this `if` matters, result is the same if we just always do the code?